Mathlib/Algebra/Category/ModuleCat/Presheaf.lean

```diff
@@ -416,3 +416,56 @@ lemma restrictionApp_toPresheafOfModules {X Y : Cᵒᵖ} (f : X ⟶ Y) :
     PresheafOfModules.restrictionApp f M.toPresheafOfModules = M.map f := rfl
 
 end BundledCorePresheafOfModules
+
+namespace PresheafOfModules
+
+variable (R)
+
+/-- Auxiliary definition for `unit`. -/
+def unitCore : CorePresheafOfModules R where
+  obj X := R.obj X
+  map {X Y} f := by
+    exact
+      { toFun := (R.map f).toFun
+        map_add' := by simp
+        map_smul' := by simp }
+
+/-- The obvious free presheaf of modules of rank `1`. -/
+abbrev unit : PresheafOfModules R := (unitCore R).toPresheafOfModules
+
+lemma unit_map_one {X Y : Cᵒᵖ} (f : X ⟶ Y) : (unit R).map f (1 : R.obj X) = (1 : R.obj Y) :=
```

Comment:
Doesn't this hold for any `PresheafOfModules`? Do we have this (as a simp lemma)?

Reply:
No, this is specific to the ring presheaf considered as a presheaf of modules over itself.

Reply:
I don't think so, because `1` only makes sense in the underlying ring.